[BUG]: CFG doesn't work well for `clean`

### Describe the bug

`Clean` assembly output looks very much like C++ output, but for some reason the CFG parses labels as functions.

### Steps to reproduce

https://www.compiler-explorer.com/z/n4KrqGhqs

### Expected behavior

proper CFG

### Reproduction link

https://www.compiler-explorer.com/z/n4KrqGhqs

### Screenshots

Not applicable

### Operating System

_No response_

### Browser version

_No response_

@mattgodbolt 

λ make EXTRA_ARGS="--language Clean"
C:/ProgramData/chocolatey/lib/make/tools/install/bin/make.exe prebuild
make[1]: Entering directory 'D:/openSource/compiler-explorer'
D:\openSource\compiler-explorer\etc\scripts\find-node: line 29: /c/Program: No such file or directory
make[1]: *** [Makefile:18: .node-bin] Error 127
make[1]: Leaving directory 'D:/openSource/compiler-explorer'
make: *** [Makefile:92: run] Error 2

not working , i tried using different solutions but can't resolve this. 
I am trying manual method right now. Please see if there is any solution regarding this.

`error : make: *** No rule to make target 'Clean'. Stop.` would be not surprising to give the same result

The makefile does not have a target called clean, that's not how the makefile works. It also doesn't install the compilers for you, it just launches the website with as well as it's possible to do so with the settings that it's configured with, but it's not magic.

You can read more about our configuration system to say where a particular compiler resides on your computer, here: https://github.com/compiler-explorer/compiler-explorer/blob/main/docs/AddingACompiler.md

To install the compilers that you need, you can use our other repository and look through this doc: https://github.com/compiler-explorer/infra/blob/main/docs/installing_compilers.md (note: also Linux only)

--- a/lib/cfg/cfg-parsers/base.ts
+++ b/lib/cfg/cfg-parsers/base.ts
@@ -214,7 +214,15 @@ export class BaseCFGParser {
         const bb = arrBB[bbIdx];
         if (hasName(asmArr, bb)) return asmArr[bb.end].text;
         const newBbName = generateName(bb.nameId, bb.end);
-        arrBB[bbIdx + 1].nameId = newBbName;
+        // bbIdx can be the last canonical basic block in the function (e.g. when a
+        // conditional jump is the final instruction and there's no fallthrough block
+        // to rename). In that case there's nothing to point the "not taken" edge at,
+        // so we still return a name for the edge but skip mutating a block that
+        // doesn't exist. The layout code already ignores edges whose target isn't a
+        // real node (see graph-layout-core.ts), so this degrades gracefully.
+        if (arrBB[bbIdx + 1]) {
+            arrBB[bbIdx + 1].nameId = newBbName;
+        }
         return newBbName;
     }
 

--- a/test/cfg-cases/cfg-base.cond-jmp-last-instruction.json
+++ b/test/cfg-cases/cfg-base.cond-jmp-last-instruction.json
@@ -1,0 +1,40 @@
+{
+  "asm": [
+    {
+      "text": "main:",
+      "source": null
+    },
+    {
+      "text": "  cmp eax, 0",
+      "source": null
+    },
+    {
+      "text": "  je .L2",
+      "source": null
+    }
+  ],
+  "cfg": {
+    "main:": {
+      "nodes": [
+        {
+          "id": "main:",
+          "label": "main:\n  cmp eax, 0\n  je .L2"
+        }
+      ],
+      "edges": [
+        {
+          "from": "main:",
+          "to": ".L2:",
+          "arrows": "to",
+          "color": "green"
+        },
+        {
+          "from": "main:",
+          "to": "main:@3",
+          "arrows": "to",
+          "color": "red"
+        }
+      ]
+    }
+  }
+}

--- a/test/cfg-tests.ts
+++ b/test/cfg-tests.ts
@@ -52,6 +52,14 @@ describe('Cfg test cases', () => {
     // TODO: Consider replacing with https://github.com/vitest-dev/vitest/issues/703
     const files = fsSync.readdirSync(testcasespath);
 
+    describe('base', () => {
+        for (const filename of files.filter(x => x.includes('cfg-base'))) {
+            it(filename, async () => {
+                await DoCfgTest('', path.join(testcasespath, filename));
+            });
+        }
+    });
+
     describe('gcc', () => {
         for (const filename of files.filter(x => x.includes('gcc'))) {
             it(filename, async () => {

